subsys/bluetooth/host/gatt.c

```diff
@@ -2024,10 +2024,9 @@ static void gatt_indicate_rsp(struct bt_conn *conn, uint8_t err,
 	}
 }
 
-static struct bt_att_req *gatt_req_alloc(bt_att_func_t func, void *params,
-					 bt_att_encode_t encode,
-					 uint8_t op,
-					 size_t len)
+static struct bt_att_req *gatt_req_alloc(bt_att_func_t func, void *params, bt_att_encode_t encode,
+					 uint8_t op, size_t len,
+					 enum bt_att_bearer_option bearer_option)
```

Comment:
i could not follow why bearer_option is needed as separate arg

Reply:
The bearer option field is not at the same offset in all the different `bt_gatt_*_params`. One thus need to know which param struct is used to get the bearer option. `att_req::att_op` can not be used to distinguish them because they are not one-to-one with the param structs. I thus added the bearer option to the `att_req` so that it could be set while the type of the param struct is known.